tests/upgrade/basic/task.yaml

```diff
@@ -21,80 +22,120 @@ execute: |
-        fedora-*|centos-*)
-            distro_purge_package snapd-selinux || true
-            ;;
-    esac
+    if [[ "$SPREAD_REBOOT" == "0" ]]; then
+
+        echo "Remove snapd and snap-confine"
+        distro_purge_package snapd snap-confine || true
+        case "$SPREAD_SYSTEM" in
+            fedora-*|centos-*)
+                distro_purge_package snapd-selinux || true
+                ;;
+        esac
+
+        echo "Install previous snapd version from the store"
+        distro_install_package snap-confine snapd
+
+        case "$SPREAD_SYSTEM" in
+            arch-*|centos-*)
+                # start the services on distros where that doesn't happen
+                # automnatically
+                systemctl daemon-reload
+                systemctl restart snapd.socket
+                systemctl enable snapd.socket
+                ;;
+        esac
+
+        # need to be seeded to allow snap install
+        if ! snap wait 2>&1|MATCH "unknown command" ; then
+            snap wait system seed.loaded
+        fi
 
-    echo "Install previous snapd version from the store"
-    distro_install_package snap-confine snapd
+        prevsnapdver=$(snap --version|grep "snapd ")
+        echo "$prevsnapdver" > prevsnapdver.out
 
-    case "$SPREAD_SYSTEM" in
-        arch-*|centos-*)
-            # start the services on distros where that doesn't happen
-            # automnatically
-            systemctl daemon-reload
-            systemctl restart snapd.socket
-            systemctl enable snapd.socket
-            ;;
-    esac
+        if [[ "$SPREAD_SYSTEM" = debian-* ]] ; then
+            # For debian we install the latest core snap independently until
+            # the bug fix is on stable once 2.27 landed
+            snap install core
+        fi
 
-    # need to be seeded to allow snap install
-    if ! snap wait 2>&1|MATCH "unknown command" ; then
-        snap wait system seed.loaded
-    fi
+        echo "Install sanity check snaps with it"
+        snap install test-snapd-tools
+        snap install test-snapd-auto-aliases
 
-    prevsnapdver=$(snap --version|grep "snapd ")
+        do_classic=no
+        if is_classic_confinement_supported ; then
+            install_local_classic test-snapd-classic-confinement
+            do_classic=yes
+            # preserve the state across reboots if necessary
+            touch do-classic
+        fi
 
-    if [[ "$SPREAD_SYSTEM" = debian-* ]] ; then
-        # For debian we install the latest core snap independently until
-        # the bug fix is on stable once 2.27 landed
-        snap install core
-    fi
+        echo "Sanity check installs"
+        test-snapd-tools.echo Hello | grep Hello
+        test-snapd-tools.env | grep SNAP_NAME=test-snapd-tools
+        test_snapd_wellknown1|MATCH "ok wellknown 1"
+        test_snapd_wellknown2|MATCH "ok wellknown 2"
+
+        context_transition=0
+        case "$SPREAD_SYSTEM" in
+            fedora-*|centos-*)
```

Comment:
What about Amazon linux?

Reply:
SELinux is not enabled there, and the EPEL package cannot be installed due to outdated dependencies in Amazon Linux repos, so there is nothing to upgrade from at the moment.